openid-connect-server/src/test/java/org/mitre/oauth2/service/impl/TestDefaultOAuth2AuthorizationCodeService

```diff
@@ -0,0 +1,135 @@
+	
+
+	@Mock
+	private AuthorizationCodeRepository repository;
+	
+	@Mock
+	private AuthenticationHolderRepository authenticationHolderRepository;
+	
+	@Mock
+	private OAuth2Authentication authentication;
+
+	@InjectMocks
+	private DefaultOAuth2AuthorizationCodeService service;
+	
+	
+
+	@Before
+	public void setUp() {
+		Mockito.reset(repository);
+		Mockito.reset(authenticationHolderRepository);
+		Mockito.when(repository.getByCode(codeNotFound)).thenReturn(null);
+		
+		codeNotFound = "codeNotFound";
+		authHolder = new AuthenticationHolderEntity();
+		authHolder.setAuthentication(authentication);
+		authenticationHolderRepository.save(authHolder);
+		entity = new AuthorizationCodeEntity();
+
+	}
+	
+
+	@Test
+	public void testCreateAuthorizationCode(){
+		
+		OAuth2Request auth2Request = Mockito.mock(OAuth2Request.class);
+		Authentication authObj = Mockito.mock(Authentication.class);
+		OAuth2Authentication authentication = new OAuth2Authentication(auth2Request, authObj);
+		String code = service.createAuthorizationCode(authentication);
+		ArgumentCaptor<AuthorizationCodeEntity> argumentCaptor = ArgumentCaptor.forClass(AuthorizationCodeEntity.class);
+		Mockito.verify(repository).save(argumentCaptor.capture());
+		assertThat(argumentCaptor.getValue(), equalTo(code));
+	
+		}
+
+	@Test
+	public void testConsumeAuthorizationCodeFound() {
+		String code = service.createAuthorizationCode(authentication);
+		assertEquals(repository.getByCode(code),authHolder);
+		when(repository.save(Matchers.isA(AuthorizationCodeEntity.class))).thenReturn(entity);
+		}
+
+	@Test
+	public void testConsumeAuthorizationCodeNotFound() {
+		assertThat(null, equalTo(service.consumeAuthorizationCode(codeNotFound)));	
+		}
+	
+	@Test(expected = InvalidGrantException.class)
+	public void testCanNotConsumeTwice() {
+		String code = service.createAuthorizationCode(authentication);
+		service.consumeAuthorizationCode(code);
+		service.consumeAuthorizationCode(code);
+		}
+
+	@Test
+	public void testConsumeAuthorizationCode(){
+
+		doNothing().when(authHolder).setAuthentication(Matchers.isA(OAuth2Authentication.class));
+		when(authRepo.save(Matchers.isA(AuthenticationHolderEntity.class))).thenReturn(authHolder);
+		}
+	
+	@Test
+	public void testRespositoryGetAndSet() {
+		
+		service.setRepository(repository);
+		assertEquals(service.getRepository(), repository);			
+		}
+	
+	@Test
+	public void testExpirationSecond() {
+
+		service.setAuthCodeExpirationSeconds(10);
+		assertEquals(service.getAuthCodeExpirationSeconds(), 10);			
+		}
+	
+	@Test
+	public void testCodeAppropiateSecond() {
+		
+		Calendar calendar = Calendar.getInstance(); 
+		calendar.add(Calendar.SECOND, 99);
+		service.setAuthCodeExpirationSeconds(100);
+		ArgumentCaptor<AuthorizationCodeEntity> argumentCaptor = ArgumentCaptor.forClass(AuthorizationCodeEntity.class);
+		Mockito.verify(repository).save(argumentCaptor.capture());
+		assertThat(argumentCaptor.capture(), is(equalTo(calendar.getTime())));
```

Comment:
As stated in previous comments, this should be doing a softer comparison than what it's doing right now. Instead of using `equalTo` make sure that it's within a 1-2 second window of the timestamp.